assets/js/SignPanel.tsx

```diff
@@ -226,199 +218,162 @@ class SignPanel extends React.Component<
     });
   }
 
-  render(): JSX.Element {
-    const {
-      alerts,
-      signId,
-      signContent,
-      currentTime,
-      line,
-      signConfig,
-      setConfigs,
-      realtimeId,
-      readOnly,
-      signGroup,
-      modes = {
-        auto: true,
-        custom: true,
-        headway: true,
-        off: true,
-      },
-    } = this.props;
-
-    const {
-      staticLine1,
-      staticLine2,
-      customChanges,
-      initialTime,
-      confirmingUngroup,
-    } = this.state;
-
-    return (
-      <section aria-label={signId}>
-        <div className="viewer--sign">
-          <div
-            className="viewer--sign-id"
-            style={{ borderTopColor: lineToColor(line) }}
-          >
-            <span style={fontSize(signId)}>{signId}</span>
-            {readOnly && (
-              <div className="viewer--mode-text">
-                {displayName(signConfig.mode)}
-              </div>
+  return (
+    <section aria-label={signId}>
+      <div className="viewer--sign">
+        <div
+          className="viewer--sign-id"
+          style={{ borderTopColor: lineToColor(line) }}
+        >
+          <span style={fontSize(signId)}>{signId}</span>
+          {readOnly && (
+            <div className="viewer--mode-text">
+              {displayName(signConfig.mode)}
+            </div>
+          )}
+          {!readOnly && !signGroup && (
+            <div>
+              <select
+                id={realtimeId}
+                className="viewer--mode-select"
+                value={signConfig.mode}
+                onChange={(event) => {
```

Comment:
This function seems complex enough that maybe it shouldn't be defined inline in the markup. (at least, if something like `handleInputLine1` gets its own function, this should too)